sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -342,6 +384,103 @@ pi_result piDeviceGetInfo(pi_device device, pi_device_info paramName,
     return PI_SUCCESS;
   }
 
+  case PI_DEVICE_INFO_PARTITION_PROPERTIES: {
+    // SYCL spec says: if this SYCL device cannot be partitioned into at least
+    // two sub devices then the returned vector must be empty.
+    pi_uint32 partitionMaxSubDevices = 0;
+    if (device->subLevel == -1)
+      device->subLevel = getSubLevel(device);
```

Comment:
This makes little sense to me. Why don't we report an error immediately for the invalid level? Also, I've realized that I'd like to see a named const/constexpr variable for "-1".

Reply:
Oops..this is a typo that seems to have snuck up at the last minute. It needs to be 1. I will replace these consts with named consts. Thanks